daprdocs/content/en/dotnet-sdk-docs/dotnet-client/_index.md

```diff
@@ -317,51 +346,24 @@ namespace LockService
 
 ## Sidecar APIs
 ### Sidecar Health
-The .NET SDK provides a way to poll for the sidecar health, as well as a convenience method to wait for the sidecar to be ready.
-
-#### Poll for health
-This health endpoint returns true when both the sidecar and your application are up (fully initialized).
-
-```csharp
-var client = new DaprClientBuilder().Build();
-
-var isDaprReady = await client.CheckHealthAsync();
-
-if (isDaprReady) 
-{
-    // Execute Dapr dependent code.
-}
-```
+While the .NET SDK provides a way to poll for the sidecar health, it is not generally recommended that developer
+utilize this functionality unless they are explicitly using Dapr to also retrieve secrets or configuration values.
 
-#### Poll for health (outbound)
-This health endpoint returns true when Dapr has initialized all its components, but may not have finished setting up a communication channel with your application.
+There are two methods available:
+- `CheckOutboundHealthAsync` which queries an outbound readiness endpoint in the Dapr Health API {{% ref health_api#wait-for-specific-health-check-against-outbound-path %}}
+for a successful HTTP status code and reports readiness based on this value.
+- `WaitForSidecarAsync` continuously polls `CheckOutboundHealthAsync` until it returns a successful status code.
 
-This is best used when you want to utilize a Dapr component in your startup path, for instance, loading secrets from a secretstore.
-
-```csharp
-var client = new DaprClientBuilder().Build();
+The "outbound" direction refers to the communication outbound from the Dapr runtime to your application. If your 
+application doesn't use actors, secret management, configuration retrieval or workflows, the runtime will not attempt
+to create an outbound connection, meaning that if your application just takes a naive dependency on `WaitForSidecarAsync`
+without any of these Dapr components, it will indefinitely lock up during startup as the endpoint will never be established.
```

Comment:
```suggestion
without using any of these Dapr components, it will indefinitely lock up during startup as the endpoint will never be established.
```